File: demo/src/main/java/com/backend/demo/entity/Receiver.java
package com.backend.demo.entity;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.OneToOne;
import javax.persistence.Table;

import com.fasterxml.jackson.annotation.JsonBackReference;

@Entity
@Table(name="receiver")
public class Receiver {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "id")
	private int id;
	
	@Column(name="name")
	private String name;
	
	@Column(name="phone_number")
	private int phoneNumber;
	
	@Column(name="address")
	private String address;
	
	@OneToOne(mappedBy = "receiver",cascade = CascadeType.ALL)
	@JsonBackReference
	private Shipment shipment; 
	
	@Column(name = "city")
	private String city;
	
	@Column(name = "district")
	private String district;
	
	@Column(name = "notes")
	private String notes;
	

	public Receiver() {}

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public int getPhoneNumber() {
		return phoneNumber;
	}

	public void setPhoneNumber(int phoneNumber) {
		this.phoneNumber = phoneNumber;
	}

	public String getAddress() {
		return address;
	}

	public void setAddress(String address) {
		this.address = address;
	}

	public Shipment getShipment() {
		return shipment;
	}

	public void setShipment(Shipment shipment) {
		this.shipment = shipment;
	}

	public String getCity() {
		return city;
	}

	public void setCity(String city) {
		this.city = city;
	}

	public String getDistrict() {
		return district;
	}

	public void setDistrict(String district) {
		this.district = district;
	}

	public String getNotes() {
		return notes;
	}

	public void setNotes(String notes) {
		this.notes = notes;
	}
	
	
}

File: delivery_agent_app/lib/models/Receiver.dart
class Receiver {
  final int? id;
  final String? name;
  final String? phoneNumber;
  final String? address;
  final String? city;
  final String? district;
  final String? notes;

  Receiver({
    required this.id,
    required this.name,
    required this.phoneNumber,
    required this.city,
    required this.address,
    required this.district,
    this.notes,
  });

  factory Receiver.fromJson(json) {
    return Receiver(
      id: json['id'],
      name: json['name'] ?? '',
      phoneNumber: json['phoneNumber'] ?? 0,
      city: json['city'] ?? '',
      district: json['district'] ?? '',
      address: json['address'] ?? '',
      notes: json['notes'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name ?? '',
      'phoneNumber': phoneNumber ?? 0,
      'city': city ?? '',
      'district': district ?? '',
      'address': address ?? '',
      'notes': notes ?? '',
    };
  }
}
